### PR TITLE
feat(app): Add Custom Trigger Example

### PR DESCRIPTION
**Pull request by AlexCuse**
Introduce an example for the custom trigger functionality introduced to
the SDK in https://github.com/edgexfoundry/app-functions-sdk-go/pull/587

## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/edgex-examples/blob/master/.github/Contributing.md.

## Issue Number:
#29

## What is the new behavior?
Introduces an example service that uses a custom trigger to read messages from stdin.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Specific Instructions
Are there any specific instructions or things that should be known prior to reviewing?

## Other information

### CODE REVIEW

File: application-services/custom/custom-trigger/main.go [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+		edgexSdk.LoggingClient.Error("MakeItRun returned error: ", err.Error())
+		os.Exit(-1)
+	}
+
+	// Do any required cleanup here
+	os.Exit(0)
+}
+
+func printToConsole(edgexcontext *appcontext.Context, params ...interface{}) (bool, interface{}) {
+	input, err := util.CoerceType(params[0])
```
Comment (lenny-intel) [inline application-services/custom/custom-trigger/main.go:128]: Since not changing target type, this function is being passed an Event, correct?

**Review by lenny-intel — CHANGES_REQUESTED** [review]

**Comment by lenny-intel** [comment]
Please add a README which show example of running the app service.

File: application-services/custom/cloud-event/main.go [hunk 2]
```diff
@@ -93,37 +93,7 @@ func gotEvent(ctx context.Context, event cloudevents.Event) {
 	var readingData string
 	event.DataAs(&readingData)
 	fmt.Printf("CloudEvent received reading value: %v\n", readingData)
-}
-
-// App function to send the cloudevent to the receiver
```
[the remark this answers is not in the document]
Comment (AlexCuse) [reply application-services/custom/cloud-event/main.go:98]: ha nope I'll knock these out real quick

File: application-services/custom/custom-trigger/main.go (hunk 1 above)
Comment (AlexCuse) [reply application-services/custom/custom-trigger/main.go:125]: Its using TargetType = &[]byte{}

**Comment by AlexCuse** [comment]
Sorry about that not sure why README didn't get picked up the first time.  Should be OK now @lenny-intel

I have had cloud events on my list of stuff to research for awhile now I'll try and clean up that service when I get a chance.

File: application-services/custom/custom-trigger/main.go (hunk 1 above)
Comment (lenny-intel) [reply application-services/custom/custom-trigger/main.go:125]: Ok, I missed that some the...

File: application-services/custom/cloud-event/main.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (lenny-intel) [reply application-services/custom/cloud-event/main.go:98]: This change needs to be un-done from this PR.

**Review by lenny-intel — APPROVED** [review]
LGTM

**Pull request merged**